sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/datasources/subscriptions/SubscriptionsDbConnector.groovy

```diff
@@ -104,70 +105,55 @@ class SubscriptionsDbConnector implements SubscriptionDataSource, SubscribedProj
           }
     }
     
-    private int getSubscriberId(Connection connection, Subscriber subscriber) {
-          int subscriberId = fetchExistingSubscriberId(subscriber)
-          if(subscriberId <= 0) {
-            String query = "INSERT INTO subscriber (first_name, last_name, email) VALUES(?, ?, ?)"
-            
-            def statement = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)
-            
+    private Optional<Integer> getSubscriberId(Subscriber subscriber) {
+        String query = "SELECT id FROM person WHERE first_name = ? AND last_name = ? AND title = ? AND email = ?"
+
+        Optional<Integer> personId = Optional.empty()
+        Connection connection = connectionProvider.connect()
+
+        connection.withCloseable {
+            def statement = connection.prepareStatement(query)
             statement.setString(1, subscriber.firstName)
             statement.setString(2, subscriber.lastName)
-            statement.setString(3, subscriber.email)
-            statement.execute()
-            def keys = statement.getGeneratedKeys()
-            while (keys.next()) {
-              subscriberId = keys.getInt(1)
+            statement.setString(3, subscriber.title)
+            statement.setString(4, subscriber.email)
+
+            ResultSet result = statement.executeQuery()
+            while (result.next()) {
+                personId = Optional.of(result.getInt(1))
             }
-          }
-          return subscriberId
+        }
+        return personId
     }
     
     private void addSubscription(Connection connection, int subscriberId, String projectCode) {
-          if (!projectAlreadySubscribed(subscriberId, projectCode)) {
-              String query = "INSERT INTO subscription (project_code, subscriber_id) VALUES(?, ?)"
+        if(subscriberId == -1){
```

Comment:
wait why `-1` ?

Reply:
Good point, that was a leftover from when the id was not an optional, will remove this part of the code